### PR TITLE
Supplement image widget docs

### PR DESCRIPTION
**Pull request by covercash2**
this is mostly just a description of the existing API with a note about the "image" feature and more links.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ You can find its changes [documented below](#060---2020-06-01).
 
 ### Docs
 
+- Added documentation for the `druid::Image` widget. ([#1018] by [@covercash2])
```
Comment (xStrom) [inline CHANGELOG.md:28]: I think just having `Image` without the qualifer is enough, that's what we've been using for types in the changelog.

File: druid/src/widget/image.rs [hunk 2]
```diff
@@ -25,7 +25,52 @@ use crate::{
...
+///
+/// Note: interpolation can lead to blurry or pixelated images and so is not
+/// recommended for things like icons. Instead consider using
+/// [SVG files](https://en.wikipedia.org/wiki/Scalable_Vector_Graphics)
+/// and enabling the `svg` feature with `cargo`.
+///
+/// (See also:
+/// [`ImageData`],
+/// [`FillStrat`],
+/// [`InterpolationMode`]
```
Comment (xStrom) [inline druid/src/widget/image.rs:41]: These links aren't defined?

File: druid/src/widget/image.rs [hunk 3]
```diff
@@ -25,7 +25,52 @@ use crate::{
...
 };
 
-/// A widget that renders an Image
+/// A widget that renders a bitmap Image.
+///
+/// Contains data about how to fill given space and interpolate pixels.
+/// Configuration options are provided via the builder pattern.
+///
+/// Note: interpolation can lead to blurry or pixelated images and so is not
+/// recommended for things like icons. Instead consider using
```
Comment (xStrom) [inline druid/src/widget/image.rs:34]: Can lead to when? With DPI scaling? Should mention just briefly that when scaling.

File: druid/src/widget/image.rs [hunk 4]
```diff
@@ -25,7 +25,52 @@ use crate::{
     PaintCtx, Rect, RenderContext, Size, UpdateCtx, Widget,
 };
 
-/// A widget that renders an Image
+/// A widget that renders a bitmap Image.
+///
+/// Contains data about how to fill given space and interpolate pixels.
```
Comment (xStrom) [inline druid/src/widget/image.rs:30]: Lack of articles feels off. Perhaps something like *How to fill **the** given space*

File: druid/src/widget/image.rs [hunk 5]
```diff
@@ -25,7 +25,52 @@ use crate::{
...
+///     widget::{Image, ImageData, FillStrat},
+///     piet::InterpolationMode,
+/// };
+///
+/// let image_data = ImageData::empty();
+/// let image_widget = Image::new(image_data)
+///     // set fill strategy
+///     .fill_mode(FillStrat::Fill)
+///     // set interpolation mode
+///     .interpolation_mode(InterpolationMode::NearestNeighbor);
```
Comment (xStrom) [inline druid/src/widget/image.rs:58]: Not sure that we want to show the example with nearest neighbor. That's rarely what is desired.

File: druid/src/widget/image.rs [hunk 6]
```diff
@@ -25,7 +25,52 @@ use crate::{
...
+/// Create an image widget and configure it using builder methods
+/// ```
+/// use druid::{
+///     widget::{Image, ImageData, FillStrat},
+///     piet::InterpolationMode,
+/// };
+///
+/// let image_data = ImageData::empty();
+/// let image_widget = Image::new(image_data)
+///     // set fill strategy
```
Comment (xStrom) [inline druid/src/widget/image.rs:55]: This and the other three inline comments seem to be missing articles. *Set **the** fill strategy*

File: druid/src/widget/image.rs [hunk 7]
```diff
@@ -35,7 +80,10 @@ pub struct Image {
 impl Image {
     /// Create an image drawing widget from `ImageData`.
     ///
-    /// The Image will scale to fit its box constraints.
+    /// By default, the Image will scale to fit its box constraints
+    /// ([`FillStrat::Fill`])
+    /// and will be scaled bilinearly
+    /// ([`InterpolationMode::Bilinear`])
```
Comment (xStrom) [inline druid/src/widget/image.rs:86]: These two links aren't defined.

File: druid/src/widget/image.rs [hunk 8]
```diff
@@ -50,7 +98,7 @@ impl Image {
         self
     }
 
-    /// Modify the widget's `FillStrat`.
+    /// Modify the widget's [`FillStrat`].
```
Comment (xStrom) [inline druid/src/widget/image.rs:101]: Link isn't defined.

File: druid/src/widget/image.rs [hunk 9]
```diff
@@ -61,7 +109,7 @@ impl Image {
         self
     }
 
-    /// Modify the widget's `InterpolationMode`.
+    /// Modify the widget's [`InterpolationMode`].
```
Comment (xStrom) [inline druid/src/widget/image.rs:112]: Link isn't defined.

File: druid/src/widget/image.rs [hunk 10]
```diff
@@ -106,7 +154,14 @@ impl<T: Data> Widget<T> for Image {
     }
 }
 
-/// Stored Image data.
+/// Loaded image data.
```
Comment (xStrom) [inline druid/src/widget/image.rs:157]: How about *Processed image data.*

File: druid/src/widget/image.rs [hunk 11]
```diff
@@ -106,7 +154,14 @@ impl<T: Data> Widget<T> for Image {
...
+///
+/// By default, Druid does not parse image data.
+/// However, [enabling the `image` feature](../index.html#optional-features)
```
Comment (xStrom) [inline druid/src/widget/image.rs:160]: Let's move the link definition to the end, like most other documentation does it.

File: druid/src/widget/image.rs [hunk 12]
```diff
@@ -106,7 +154,14 @@ impl<T: Data> Widget<T> for Image {
...
+/// provides several
+/// methods by which you can load image files.
+///
+/// Contains raw bytes, dimensions, and image format ([`druid::piet::ImageFormat`]).
```
Comment (xStrom) [inline druid/src/widget/image.rs:164]: This link isn't defined. Also it should just say `piet::ImageFormat`.

File: druid/src/widget/image.rs (hunk 2 above)
Comment (covercash2) [reply druid/src/widget/image.rs:41]: they don't have to be when `rustdoc` can figure them out (i haven't looked into the scope of this in detail). these links work as expected, AFAICT, but it's not problem if they need to be updated to match the more bibliographical approach in the other docs.

File: druid/src/widget/image.rs (hunk 5 above)
Comment (covercash2) [reply druid/src/widget/image.rs:58]: right. it's part of my particular use case (texture packer) so maybe i was a little too fixated on it.

easily fixed!

File: druid/src/widget/image.rs (hunk 10 above)
Comment (covercash2) [reply druid/src/widget/image.rs:157]: definitely, i struggled with this a bit. i was hoping to get a correction for this one, frankly.

File: druid/src/widget/image.rs (hunk 3 above)
Comment (covercash2) [reply druid/src/widget/image.rs:34]: i'm not sure exactly how DPI scaling is relevant here? like images may look fine on a normal DPI screen but blurry when in hiDPI situations?

File: druid/src/widget/image.rs (hunk 2 above)
Comment (xStrom) [reply druid/src/widget/image.rs:41]: The links certainly don't work with stable rust. I know that intra-doc rustdoc links are [on their way to stabilization](https://github.com/rust-lang/rust/issues/43466) but they aren't here yet. Our docs need to work with stable rust.

Comment (covercash2) [reply druid/src/widget/image.rs:41]: ah, that's the missing piece. sorry about that.

**Pull request merged**